### PR TITLE
drivers/camera: Support the private data for imgsensor and imgdata

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
Make video driver model close other driver framework and support the multiple instances from one driver.

## Impact
video capture

## Testing

sim:camera and spresense:camera

### CODE REVIEW

**Comment by acassis** [comment]
@masayuki2009 @jerpelea does Spresence camera still working after this modification?

File: include/nuttx/video/imgsensor.h [hunk 1]
```diff
@@ -265,7 +294,7 @@ typedef struct imgsensor_capability_range_s
 typedef struct imgsensor_capability_discrete_s
 {
   int8_t  nr_values;
-  int32_t *values;
+  const int32_t *values;
```
Comment (pkarashchenko) [inline include/nuttx/video/imgsensor.h:297]:
```suggestion
  FAR const int32_t *values;
```

**Review by pkarashchenko — COMMENTED** [review]
In general looks good. Just need a feedback from @SPRESENSE and team if that still works because the last feedback I had in https://github.com/sonydevworld/spresense/pull/536#issuecomment-1386728620 was that things got broken and I'm not sure if those have been fixed till now.

File: include/nuttx/video/imgsensor.h (hunk 1 above)
Comment (xiaoxiang781216) [reply include/nuttx/video/imgsensor.h]: Done.

**Comment by SPRESENSE** [comment]
LGTM

**Review by GUIDINGLI — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]

**Pull request merged**